src/assets/custom/js/websiteHeader.js

```diff
@@ -31,42 +31,59 @@ var websiteHeader = function () {
 
   function handleScrollPosition(latestWindowPosition) {
     var upwardMovement =
-      previousWindowPosition > latestWindowPosition + tolerance;
+      previousWindowPosition > latestWindowPosition + scrollTolerance;
     var downwardMovement =
-      previousWindowPosition < latestWindowPosition - tolerance;
+      previousWindowPosition < latestWindowPosition - scrollTolerance;
 
-    var subMenuOpen = HEADER.classList.contains(HEADER_HAS_OPEN_SUBMENU_CLASS);
+    var subMenuOpen = header.classList.contains(
+      'website-header--has-open-submenu'
+    );
 
-    if (atTheTop(latestWindowPosition) || (downwardMovement && !subMenuOpen)) {
-      restoreNaturalPosition();
+    if (
+      startingPositionIsInView(latestWindowPosition) ||
+      atTheTop(latestWindowPosition) ||
+      upwardMovement
+    ) {
+      revealHeader();
       return;
     }
 
-    if (startingPositionIsInView(latestWindowPosition)) {
-      return;
-    }
+    if (downwardMovement && !websiteHeaderOpen()) {
+      if (subMenuOpen) {
+        closeSubMenu =
+          window.__CODURANCE.websiteNavigation.closeOpenSubMenu ||
+          function () {};
+        closeSubMenu();
+      }
```

Comment:
```suggestion
      if(
        window.__CODURANCE && 
        window.__CODURANCE.websiteNavigation && 
        window.__CODURANCE.websiteNavigation.closeOpenSubMenu
      ) {
        window.__CODURANCE.websiteNavigation.closeOpenSubMenu();
       }
```

Personally I wouldn't assume that `window.__CODURANCE.websiteNavigation.closeOpenSubMenu` is defined. 

Also, that function already handles whether or not a subMenu is open so you can remove `subMenuOpen` from here and potentially from this whole file. Which would be nice.

Reply:
This is a lot better and cleaner too. It's now been implemented here: https://github.com/codurance/site/pull/1940/commits/bbea935a1c2a48d0f6f53ee09248e47647187d59